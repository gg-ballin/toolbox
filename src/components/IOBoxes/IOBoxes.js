import React from 'react';
import './IOBoxes.scss';


const IOBoxes = ({ searchChange, onButtonSubmit, output }) => {
  return  (
    
    <div className='pa2'>
      <label>
        <input
          className='pa3 ba b--black bg-white'
          type='search'
          placeholder='Input'
          onChange={searchChange}
        />
      </label>
      <br></br>
        <button
          className='w-30 grow f4 link ph3 pv2 dib white bg-red'
          onClick={onButtonSubmit}>
              âž¡
        </button>
      <label>
        <input
          className='pa3 ba b--black bg-white'
          type='text'
          placeholder='Output'
          onChange ={output}
        />
      </label>
    </div>
  );
}

export default IOBoxes;