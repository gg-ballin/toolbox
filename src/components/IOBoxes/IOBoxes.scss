.dropbtn {
    background-color: rgb(243, 38, 38);
    color: white;
    padding: 16px;
    font-size: 16px;
    border: none;
    cursor: pointer;
}