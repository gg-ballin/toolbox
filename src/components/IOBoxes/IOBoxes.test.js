import React from 'react';
import { shallow } from 'enzyme';
import IOBoxes from './IOBoxes'

it('expect to render SearchBox component', () => {
    expect(shallow(<IOBoxes /> )).toMatchSnapshot();
})
 