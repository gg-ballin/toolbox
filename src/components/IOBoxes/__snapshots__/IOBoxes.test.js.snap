// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render SearchBox component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IOBoxes />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          <input
            className="pa3 ba b--black bg-white"
            placeholder="Input"
            type="search"
          />
        </label>,
        <br />,
        <button
          className="w-30 grow f4 link ph3 pv2 dib white bg-red"
        >
          ➡
        </button>,
        <label>
          <input
            className="pa3 ba b--black bg-white"
            placeholder="Output"
            type="text"
          />
        </label>,
      ],
      "className": "pa2",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="pa3 ba b--black bg-white"
            placeholder="Input"
            type="search"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "pa3 ba b--black bg-white",
            "onChange": undefined,
            "placeholder": "Input",
            "type": "search",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "➡",
          "className": "w-30 grow f4 link ph3 pv2 dib white bg-red",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "➡",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="pa3 ba b--black bg-white"
            placeholder="Output"
            type="text"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "pa3 ba b--black bg-white",
            "onChange": undefined,
            "placeholder": "Output",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            <input
              className="pa3 ba b--black bg-white"
              placeholder="Input"
              type="search"
            />
          </label>,
          <br />,
          <button
            className="w-30 grow f4 link ph3 pv2 dib white bg-red"
          >
            ➡
          </button>,
          <label>
            <input
              className="pa3 ba b--black bg-white"
              placeholder="Output"
              type="text"
            />
          </label>,
        ],
        "className": "pa2",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="pa3 ba b--black bg-white"
              placeholder="Input"
              type="search"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "pa3 ba b--black bg-white",
              "onChange": undefined,
              "placeholder": "Input",
              "type": "search",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "➡",
            "className": "w-30 grow f4 link ph3 pv2 dib white bg-red",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "➡",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="pa3 ba b--black bg-white"
              placeholder="Output"
              type="text"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "pa3 ba b--black bg-white",
              "onChange": undefined,
              "placeholder": "Output",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
